src/dataflow/src/sink/avro_ocf.rs

```diff
@@ -9,87 +9,73 @@
 
 use std::fs::OpenOptions;
 
-use differential_dataflow::hashable::Hashable;
+use differential_dataflow::Collection;
+
+use itertools::repeat_n;
 use log::error;
-use timely::dataflow::channels::pact::Exchange;
+use timely::dataflow::channels::pact::Pipeline;
 use timely::dataflow::operators::generic::Operator;
-use timely::dataflow::{Scope, Stream};
+use timely::dataflow::Scope;
 
 use dataflow_types::AvroOcfSinkConnector;
 use expr::GlobalId;
-use interchange::avro::{DiffPair, Encoder};
-use repr::{Diff, RelationDesc, Row, Timestamp};
+use interchange::avro::{encode_datums_as_avro, Encoder};
+use mz_avro::{self};
+use repr::{RelationDesc, Row, Timestamp};
 
 pub fn avro_ocf<G>(
-    stream: &Stream<G, (Row, Timestamp, Diff)>,
+    stream: Collection<G, (Option<Row>, Option<Row>)>,
```

Comment:
minor but important: naming an instance of `Collection` `stream` makes things super confusing later. Please reconsider. Also maybe consider having this be just a `Collection<Row>` (and `map` the collection` before passing it to this func) because it seems like ocfs cant have keys and must have values and I feel like its better to enforce with the type system as much as possible

Reply:
yeah you're absolutely right, this is just stray; I'll rename it.

The reason I don't map this upstream and enforce it in the type system is because I wanted to avoid having connector-specific logic upstream of this, as much as possible.

Reply:
then i think it makes sense to do the map and pass it to a function called `avro_ocf_inner` that does the actual handling logic on the correct types?

Reply:
Do you like the current version better? I do the map at the beginning, and then the type system validates that we are using the correct types later on in the differential logic.